src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ProjectPropertiesProviders/ImplicitProjectPropertiesProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.ProjectPropertiesProviders
+{
+    /// <summary>
+    /// Provides project properties that normally should not live in the project
+    /// file but may be written from an external source (e.g. the solution file).
+    /// This provider avoids writing these properties to the project file, but if
+    /// they are already present there, it updates the value to keep in sync with
+    /// the external source.
+    /// </summary>
+    [Export("Implicit", typeof(IProjectPropertiesProvider))]
+    [Export(typeof(IProjectPropertiesProvider))]
+    [ExportMetadata("Name", "Implicit")]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+    internal class ImplicitProjectPropertiesProvider : DelegatedProjectPropertiesProviderBase
+    {
+        [ImportingConstructor]
+        public ImplicitProjectPropertiesProvider(
+            [Import("Microsoft.VisualStudio.ProjectSystem.ProjectFile")] IProjectPropertiesProvider provider)
```

Comment:
I'll keep this in the subclass following Dave's recommendation. Since more of us are using these names, I'll also port the relevant constants over from ExportContractNames (since they are not accessible here)

---
In reply to: [69321613](https://github.com/dotnet/roslyn-project-system/pull/283#discussion_r69321613) [](ancestors = 69321613)